### PR TITLE
Implementation for streaming applier recovery

### PR DESCRIPTION
**Pull request by temeo**
Implemented service calls to recover streaming appliers when requested by wsrep-lib.

### CODE REVIEW

File: sql/wsrep_schema.cc [hunk 1]
```diff
@@ -1247,6 +1269,9 @@ int Wsrep_schema::recover_sr_transactions()
       Wsrep_schema_impl::scan(frag_table, 2, seqno_ll);
       wsrep::seqno seqno(seqno_ll);
 
+      /* This is possible if the server crashes between inserting the
+         fragment into table and updating the fragment seqno after
+         certification. */
       if (seqno.is_undefined())
       {
         Wsrep_schema_impl::delete_row(frag_table);
```
Comment (sciascid) [inline sql/wsrep_schema.cc]: At this point the table is opened for read, not sure we are allowed to delete the row.
It is probably sufficient to open the table for write as well.

**Comment by temeo** [comment]
This PR shows for some reason commit e26a8ce2bb60dd09cdd00947046db28c265578f3 which already exists in bb-10.4-galera4. Will open a new PR later if it does not eventually disappear.

**Comment by temeo** [comment]
There are many test failures which didn't happen for MTRs running for other PRs at the same time. Will rerun to see if the tests were random failures.

Galera MTR results:

https://jenkins.galeracluster.com:8443/job/mtr-galera-4.x-mariadb-10.4/385/console

Completed: Failed 18/367 tests, 95.10% were successful.

Failing test(s): galera.galera_gcache_recover_full_gcache galera.galera_sst_mysqldump galera_3nodes.galera_gtid_2_cluster galera_3nodes.galera_pc_weight galera_3nodes.galera_var_dirty_reads2 galera_3nodes.galera_ist_gcache_rollover galera_sr.galera_sr_ddl_master galera_sr.galera_sr_large_fragment galera.galera#505 galera.galera_FK_duplicate_client_insert galera_sr.GCF-437 galera_sr.galera_sr_rollback_statement galera_sr.galera_sr_table_contents galera_sr.mysql-wsrep-features#165 galera_sr.galera_sr_mysqldump_sst galera_sr.galera_sr_cc_master galera_sr.galera_sr_load_data_splitting galera_sr.galera_sr_many_fragments

**Comment by temeo** [comment]
test_pr

**Comment by temeo** [comment]
New job: https://jenkins.galeracluster.com:8443/job/mtr-galera-4.x-mariadb-10.4/387/console

**Comment by temeo** [comment]
https://jenkins.galeracluster.com:8443/job/mtr-galera-4.x-mariadb-10.4/398/console

**Comment by temeo** [comment]
test_pr

**Comment by temeo** [comment]
This PR is superseded by https://github.com/codership/mariadb-server/pull/58, closing this one.

**Pull request closed without merging**